git rev-list in has_new_commits() crashes session on missing ref

## Bug

`has_new_commits()` in `agent_fox/workspace/git.py:313-327` calls `run_git(["rev-list", "--count", ...])` **without `check=False`**, so any non-zero exit code raises `WorkspaceError` and crashes the session runner.

Git exit code 128 means a ref wasn't found — this happens when the feature branch ref isn't fully visible to git immediately after worktree creation. The error surfaces as:

```
[ERROR] agent_fox.engine.session_lifecycle: Session runner failed for 100_maintainer_archetype:0 (attempt 1): git rev-list failed (exit code 128)
```

The retry (attempt 2) succeeds because by then the refs are visible. The bug is non-blocking thanks to the retry mechanism, but it wastes a full session attempt (~minutes of wall time + API cost).

## Root Cause

```python
# git.py:323-327
_rc, stdout, _stderr = await run_git(
    ["rev-list", "--count", f"{base}..{branch}"],
    cwd=repo_path,
)  # <-- no check=False → raises WorkspaceError on exit 128
```

Compare with `develop.py:106` which does the exact same `rev-list` call but correctly uses `check=False`:

```python
# develop.py:106
_rc, remote_ahead_str, _stderr = await run_git(
    ["rev-list", "--count", "develop..origin/develop"],
    cwd=repo_root,
    check=False,  # ← handles missing refs gracefully
)
```

## Observed Impact

- Seen during an orchestration run processing specs 100, 101, 102.
- `100_maintainer_archetype:0` failed attempt 1, succeeded on retry #2.
- `101_knowledge_onboarding:1:reviewer:audit-review` also failed attempt 1 with the same error.

## Proposed Fix

`has_new_commits()` should use `check=False` and return `False` when the command fails (missing ref = no new commits), matching the pattern used in `develop.py`.

```python
async def has_new_commits(repo_path: Path, branch: str, base: str) -> bool:
    validate_ref_name(branch)
    validate_ref_name(base)
    rc, stdout, _stderr = await run_git(
        ["rev-list", "--count", f"{base}..{branch}"],
        cwd=repo_path,
        check=False,
    )
    if rc != 0:
        return False
    return int(stdout.strip()) > 0
```

--- a/agent_fox/workspace/git.py
+++ b/agent_fox/workspace/git.py
@@ -320,10 +320,13 @@ async def has_new_commits(
     """
     validate_ref_name(branch)
     validate_ref_name(base)
-    _rc, stdout, _stderr = await run_git(
+    rc, stdout, _stderr = await run_git(
         ["rev-list", "--count", f"{base}..{branch}"],
         cwd=repo_path,
+        check=False,
     )
+    if rc != 0:
+        return False
     return int(stdout.strip()) > 0
 
 

--- a/tests/unit/workspace/test_has_new_commits.py
+++ b/tests/unit/workspace/test_has_new_commits.py
@@ -1,0 +1,54 @@
+"""Tests for has_new_commits().
+
+Regression test for GitHub issue #338: git rev-list in has_new_commits()
+crashes session on missing ref instead of returning False.
+"""
+
+from __future__ import annotations
+
+from pathlib import Path
+from unittest.mock import AsyncMock, patch
+
+import pytest
+
+from agent_fox.workspace.git import has_new_commits
+
+
+class TestHasNewCommitsMissingRef:
+    """Issue #338: has_new_commits returns False on missing ref instead of crashing."""
+
+    @pytest.mark.asyncio
+    async def test_returns_false_on_nonzero_exit(self) -> None:
+        """When git rev-list fails (exit 128), return False instead of raising."""
+        with patch(
+            "agent_fox.workspace.git.run_git",
+            new_callable=AsyncMock,
+            return_value=(128, "", "fatal: bad revision 'develop..feature/x'"),
+        ):
+            result = await has_new_commits(Path("/repo"), "feature/x", "develop")
+
+        assert result is False
+
+    @pytest.mark.asyncio
+    async def test_returns_true_when_commits_ahead(self) -> None:
+        """Normal case: branch has commits ahead of base."""
+        with patch(
+            "agent_fox.workspace.git.run_git",
+            new_callable=AsyncMock,
+            return_value=(0, "3\n", ""),
+        ):
+            result = await has_new_commits(Path("/repo"), "feature/x", "develop")
+
+        assert result is True
+
+    @pytest.mark.asyncio
+    async def test_returns_false_when_no_commits_ahead(self) -> None:
+        """Branch is even with base — no new commits."""
+        with patch(
+            "agent_fox.workspace.git.run_git",
+            new_callable=AsyncMock,
+            return_value=(0, "0\n", ""),
+        ):
+            result = await has_new_commits(Path("/repo"), "feature/x", "develop")
+
+        assert result is False

